routers/api/actions/artifacts.go

```diff
@@ -257,8 +257,11 @@ func (ar artifactRoutes) uploadArtifact(ctx *ArtifactContext) {
 		return
 	}
 
-	// update artifact size if zero
-	if artifact.FileSize == 0 || artifact.FileCompressedSize == 0 {
+	// update artifact size if zero or not match, over write artifact size
+	if artifact.FileSize == 0 ||
```

Comment:
noop. It overwrites in old logic. If same size, it can updates db record with new storage path of artifact. If not same size, it occurs error and doesn't update.